docs(llm_observability): document audio_parts and image_parts on messages (#38612)
DataDog/documentation: content/en/llm_observability/instrumentation/api.md

| tool_id | string | Unique identifier matching the corresponding tool call. |
| type | string | The type of tool result. |

#### AudioPart

> An audio segment on a message. Provide exactly one of `content` or `attachment_key`.

Reviewer: ```suggestion
An audio segment on a message. Provide either `content` or `attachment_key`.
```

| Field | Type | Description |
|-------|------|-------------|

> | mime_type [*required*] | string | The media type of the audio, for example `audio/wav` or `audio/pcm`. |

Reviewer: ```suggestion
| mime_type [*required*] | string | The media type of the audio, such as `audio/wav` or `audio/pcm`. |
```

> | content | string | The base64-encoded audio, carried inline on the message. |

Reviewer: ```suggestion
| content | string | The base64-encoded audio, carried inline with the message. |
```

> | attachment_key | string | A reference to audio stored outside the span payload, in place of inline `content`. |

Reviewer: ```suggestion
| attachment_key | string | A reference to audio stored outside the span payload, instead of inline `content`. |
```

#### ImagePart

> An image on a message. Provide exactly one of `content` or `attachment_key`.

Reviewer: ```suggestion
An image on a message. Provide either `content` or `attachment_key`.
```

| Field | Type | Description |
|-------|------|-------------|

> | mime_type [*required*] | string | The media type of the image, for example `image/png` or `image/jpeg`. |

Reviewer: ```suggestion
| mime_type [*required*] | string | The media type of the image, such as `image/png` or `image/jpeg`. |
```

> | content | string | The base64-encoded image, carried inline on the message. |

Reviewer: ```suggestion
| content | string | The base64-encoded image, carried inline with the message. |
```

> | attachment_key | string | A reference to an image stored outside the span payload, in place of inline `content`. |

Reviewer: ```suggestion
| attachment_key | string | A reference to an image stored outside the span payload, instead of inline `content`. |
```
